Add versioned docs (#486)
sky-uk/kfp-operator: docs-gen/README.md

```

This will populate the [/docs](/docs) directory, which will be served on [GitHub Pages](https://sky-uk.github.io/kfp-operator).

## Versioning
To document different versions of the KFP Operator, we store an archive of the `content/docs/` folder for each release version, under `content/archive/<version>/`.

To create a new archive version:

> 1. run `make archive-version VERSION=<new_version>` with the new version i.e. `v0.6.0`
> 8. Rebuild the site with `make build`.

Reviewer: ```suggestion
1. run `make archive-version VERSION=<new_version>` with the new version i.e. `v0.6.0`
2. Rebuild the site with `make build`.
```

> 1. run `make archive-version VERSION=<new_version>` with the new version i.e. `v0.6.0`
> 8. Rebuild the site with `make build`.

Reviewer: ```suggestion
1. run `make archive-version VERSION=<new_version>` with the new version e.g. `v0.6.0`
2. Rebuild the site with `make build`.
```

Reviewer 2: updated in [c53ea1b](https://github.com/sky-uk/kfp-operator/pull/486/commits/c53ea1bfd7c2c7e2961b6a36a0573a337fd98f17)

```

This will populate the [/docs](/docs) directory, which will be served on [GitHub Pages](https://sky-uk.github.io/kfp-operator).

## Versioning

> To document different versions of the KFP Operator, we store an archive of the `content/docs/` folder for each release version, under `content/archive/<version>/`.

Reviewer: ```suggestion
To document different versions of the KFP Operator, we store an archive of the `content/docs/` folder for each release version, under `content/versions/<version>/`.
```

To create a new version:

> Run `make archive-version VERSION=<new_version> && make build` where `new_version` is the version, e.g. `v0.6.0`

Reviewer 3: Is it worth adding the call to build within the archive-version target?

Reviewer 2: Added comments in [181ae9f](https://github.com/sky-uk/kfp-operator/pull/486/commits/181ae9f4fd5a01db5224fc48a22b9012dccd4164)
